### PR TITLE
Implement minor fixes on declarations

### PR DESCRIPTION
**Pull request by keitheous**
## Description
Minor fixes on vars, lets, consts. I thought it would be good to have no more vars, just lets and consts for consistency.

## Implementation notes
Can get to all Vars -> Lets
Can't get to all Lets which are suppose to be Const. Pretty minor, should be fine.
Question if, when we scale next time. Will our future developers forget the original context these variables?
While implementing, I ask myself if the variable is used outside of scope or not.
While implementing, I ask myself if the variable can be reassigned. If Yes, Let. If No, Const.

## Risks
* if something is suppose to be let instead of const, things will break haha

### CODE REVIEW

File: src/javascript/app.js [hunk 1]
```diff
@@ -196,6 +196,7 @@ var App = {
   },
 
   displayHome: function (data) {
+    // ask about data. Exist in method, however not during function calls
```
Comment (keitheous) [inline src/javascript/app.js]: @vaevictis - I will come ask you about this later tomorrow. =)

File: lib/javascript/base_app.js [hunk 2]
```diff
@@ -161,7 +161,7 @@ BaseApp.prototype = {
     if (!_.isFunction(fn)) {
       throw new Error('`promise` needs to be passed a Function')
     }
-    var dfd = $.Deferred()
+    let dfd = $.Deferred()
```
Comment (yu-tian113) [inline lib/javascript/base_app.js]:
```suggestion
    const dfd = $.Deferred()
```

File: src/javascript/app.js [hunk 3]
```diff
@@ -600,8 +601,8 @@ var App = {
   },
 
   serializeRequesterAttributes: function () {
-    var requesterType = $('select[name=requester_type]').val()
-    var attrs = {
+    const requesterType = $('select[name=requester_type]').val()
+    let attrs = {
```
Comment (yu-tian113) [inline src/javascript/app.js]:
```suggestion
    const attrs = {
```

File: src/javascript/app.js [hunk 4]
```diff
@@ -641,7 +642,7 @@ var App = {
         if (data.next_page) {
           makeRequest()
         } else {
-          var returnObj = {}
+          let returnObj = {}
```
Comment (yu-tian113) [inline src/javascript/app.js]:
```suggestion
          const returnObj = {}
```

File: src/javascript/app.js [hunk 5]
```diff
@@ -651,7 +652,7 @@ var App = {
         fail(params)
       }
 
-      var makeRequest = function () {
+      let makeRequest = function () {
```
Comment (yu-tian113) [inline src/javascript/app.js]:
```suggestion
      const makeRequest = function () {
```

Comment (keitheous) [reply src/javascript/app.js]: 👍

File: src/javascript/app.js (hunk 1 above)
Comment (keitheous) [reply src/javascript/app.js]: As discussed in person, this is actually part of the Ajax call made. This parameter is event

File: src/javascript/app.js (hunk 5 above)
Comment (keitheous) [reply src/javascript/app.js]: 👍

File: src/javascript/app.js (hunk 4 above)
Comment (keitheous) [reply src/javascript/app.js]: 👍

File: src/javascript/app.js (hunk 3 above)
Comment (keitheous) [reply src/javascript/app.js]: 👍

File: lib/javascript/base_app.js (hunk 2 above)
Comment (keitheous) [reply lib/javascript/base_app.js]: 👍

**Comment by keitheous** [comment]
@vaevictis Unfortunately, I did not. Upon doing it, I realised that there are quite a few reds to be fixed. =(

I'll wait for your current changes to go up, merge your changes and see how we go.

File: lib/javascript/base_app.js (hunk 2 above)
Comment (keitheous) [reply lib/javascript/base_app.js]: 👍

**Comment by vaevictis** [comment]
Hey @keitheous you can now rebase this branch against `master`.

**Pull request merged**